shachain/element_test.go

```diff
@@ -0,0 +1,179 @@
+package shachain
+
+import (
+	"reflect"
+	"testing"
+)
+
+// bitsToUint64 is a helper function which takes 'n' last bits as input and
+// create uint64 value.
+// Example:
+//  Input: 0,1,1,0,0
+//  Output: 0b000000000000000000000000000000000000000[01100] == 12
+func bitsToUint64(bs ...uint64) uint64 {
+	if len(bs) > 64 {
+		panic("number of elements should be lower then 64")
+	}
+
+	var res uint64 = 0
+	for i, e := range bs {
+		if e != 1 && e != 0 {
+			panic("wrong element, should be '0' or '1'")
```

Comment:
Alternatively, you can move the creation of the `deriveIndexTests` to within `TestDeriveIndex`. Then you'd pass in the `*testing.T` and fail the test upon an invariant violation.